ArduPlane/GCS_Mavlink.cpp

```diff
@@ -929,35 +929,33 @@ MAV_RESULT GCS_MAVLINK_Plane::handle_command_int_guided_slew_commands(const mavl
             return MAV_RESULT_DENIED;
         }
 
-         // the requested alt data might be relative or absolute
+        Location::AltFrame new_target_alt_frame;
+
         float new_target_alt = packet.z * 100;
-        float new_target_alt_rel = packet.z * 100 + plane.home.alt;
-
-         // only global/relative/terrain frames are supported
-        switch(packet.frame) {
-            case MAV_FRAME_GLOBAL_RELATIVE_ALT: {
-                if   (is_equal(plane.guided_state.target_alt,new_target_alt_rel) ) { // compare two floats as near-enough
-                    // no need to process any new packet/s with the same ALT any further, if we are already doing it.
-                    return MAV_RESULT_ACCEPTED;
-                }
-                plane.guided_state.target_alt = new_target_alt_rel;
-                break;
+        if (mavlink_coordinate_frame_to_location_alt_frame((MAV_FRAME)packet.frame, new_target_alt_frame)) {
+
+             // the requested alt data might be relative (to home)
+            if (new_target_alt_frame == Location::AltFrame::ABOVE_HOME) {
```

Comment:
This should definitely still be a `switch` statement.

We should be rejecting packets if the new frame isn't one of ABOVE_HOME or ABSOLUTE

Reply:
Interestingly, using mavlink_coordinate_frame_to_location_alt_frame() probably makes it work with ABOVE_TERRAIN "for free". Hence the changes I'm trying to make to the autotest to test that.